sdk/common/src/main/java/io/opentelemetry/sdk/resources/ResourcesConfig.java

```diff
@@ -0,0 +1,142 @@
+ * environment variables, or {@link Properties} objects.
+ *
+ * <p>For system properties and {@link Properties} objects, {@link ResourcesConfig} will look for
+ * the following names:
+ *
+ * <ul>
+ *   <li>{@code otel.java.disable.resources.providers}: to set the ResourceProvider service
+ *       providers found on the classpath to be disabled.
+ * </ul>
+ *
+ * <p>For environment variables, {@link ResourcesConfig} will look for the following names:
+ *
+ * <ul>
+ *   <li>{@code OTEL_JAVA_DISABLE_RESOURCES_PROVIDERS}: to set the ResourceProvider service
+ *       providers found on the classpath to be disabled.
+ * </ul>
+ */
+@AutoValue
+@Immutable
+public abstract class ResourcesConfig {
+  // These values are the default values for all the global parameters.
+  private static final ImmutableSet<String> OTEL_JAVA_DISABLE_RESOURCES_PROVIDERS =
+      ImmutableSet.of();
+
+  /**
+   * Returns the default {@code ResourcesConfig}.
+   *
+   * @return the default {@code ResourcesConfig}.
+   * @since 0.9.0
+   */
+  public static ResourcesConfig getDefault() {
+    return DEFAULT;
+  }
+
+  private static final ResourcesConfig DEFAULT = ResourcesConfig.newBuilder().build();
+
+  /**
+   * Returns the ResourceProvider service providers found on the classpath to be disabled.
+   *
+   * @return the ResourceProvider service providers found on the classpath to be disabled.
+   */
+  public abstract Set<String> getDisabledResourceProviders();
+
+  /**
+   * Returns a new {@link Builder}.
+   *
+   * @return a new {@link Builder}.
+   */
+  public static Builder newBuilder() {
+    return new AutoValue_ResourcesConfig.Builder()
+        .setDisabledResourceProviders(OTEL_JAVA_DISABLE_RESOURCES_PROVIDERS);
+  }
+
+  /**
+   * Returns a {@link Builder} initialized to the same property values as the current instance.
+   *
+   * @return a {@link Builder} initialized to the same property values as the current instance.
+   */
+  public abstract Builder toBuilder();
+
+  /** Builder for {@link ResourcesConfig}. */
+  @AutoValue.Builder
+  public abstract static class Builder extends ConfigBuilder<Builder> {
+
+    private static final String OTEL_JAVA_DISABLE_RESOURCES_PROVIDERS =
+        "otel.java.disable.resources.providers";
+
+    Builder() {}
+
+    /**
+     * Sets the configuration values from the given configuration map for only the available keys.
+     *
+     * @param configMap {@link Map} holding the configuration values.
+     * @return this
+     */
+    @VisibleForTesting
+    @Override
+    protected Builder fromConfigMap(
+        Map<String, String> configMap, NamingConvention namingConvention) {
+      configMap = namingConvention.normalize(configMap);
+
+      String stringValue = getStringProperty(OTEL_JAVA_DISABLE_RESOURCES_PROVIDERS, configMap);
+      if (stringValue != null) {
+        this.setDisabledResourceProviders(ImmutableSet.copyOf(stringValue.split(",")));
```

Comment:
Use a Guava `Splitter` and do `omitEmptyValues` and `trimResults`

https://guava.dev/releases/23.0/api/docs/com/google/common/base/Splitter.html